design summary (#64)
SW705E19/report: sections/design/design-summary.tex

\section{Summary of the design chapter}
In the preceding chapter we discussed aspects of the system's design.
To give the developers a sense of the visual design, we created prototypes for the different parts of the system.
A class diagram was defined, giving an understanding of the design of the system and its constituent parts.
The architecture of the system was also defined through an architecture diagram, showing how the different components of the system interface.
We researched the difference between SPAs and MPAs, concluding that the major difference is in how they build applications.
A SPA application consists of a single page that updates based on what is needed, and an MPA consists of multiple pages.
SPAs are faster when loading since only changes need to be loaded, MPAs are better for search engine optimization and have unlimited scalability.
We chose to build a SPA, since the faster loading was inline with the goals of the project. 

> We researched React, a JavaScript library, and determined it is a popular choice for frontend development with an extensive ecosystem.

Reviewer: ```suggestion
We researched React, a JavaScript library, and determined it is a popular choice for front end development with an extensive ecosystem.
```

A short introduction to react was provided, showing variables, components and states.
Finally, we considered the design of the API.
We chose to use TypeScript for implementation, and defined how an API receives a request for information, fetches the information from the data source, and returns it to the client requesting it.
For the database layer we researched relational and non-relational databases.
Relational databases scale vertically, and non-relational scale more horizontally.

> When atomicity, consistency, isolation and durability are important a relational database should be considered.

Reviewer: ```suggestion
When atomicity, consistency, isolation and durability (ACID) are important a relational database should be considered.
```

We chose to use a relational database since ACID compliance is an important part of the system.

> Finally, we defined an ERD-diagram to give a sense of the structure of the database.

Reviewer: ```suggestion
Finally, we defined an ER diagram to give a sense of the structure of the database.
```

\section{Summary of the design chapter}
In the preceding chapter we discussed aspects of the system's design.

> To give the developer's a sense of the visual design, we created prototypes for the different parts of the system.

Reviewer: ```suggestion
To give the developers a sense of the visual design, we created prototypes for the different parts of the system.
```

A SPA application consists of a single page that updates based on what is needed, and an MPA consists of multiple pages.
SPAs are faster when loading since only changes need to be loaded, MPAs are better for search engine optimization and have unlimited scalability.
We chose to build a SPA, since the faster loading was inline with the goals of the project. 
We researched React, a JavaScript library, and determined it is a popular choice for frontend development with an extensive ecosystem.
A short introduction to react was provided, showing variables, components and states.
Finally, we considered the design of the API.
We chose to use TypeScript for implementation, and defined how an API receives a request for information, fetches the information from the data source, and returns it to the client requesting it.
For the database layer we researched relational and non-relational databases.
When atomicity, consistency, isolation and durability are important a relational database should be considered.
Relational databases scale vertically, and non-relational scale more horizontally.

> We chose to use a relational database since ACID compliance is an important part of the system.

Reviewer: slå det om ACID sammen til en

> We chose to use Postgres because of experience.

Reviewer: ```suggestion
```
